### PR TITLE
DS Picker: Make responsive the modal for smaller devices

### LINKED ISSUE #67948: Dashboards: Padding not respected when scroll in `y` in the New DS Picker modal
### 
# What went wrong?

**What happened**:

  -  Because of the fact the div children container has the padding and the scroll behavior, if the content is scrolleable (mostly in mobile), the padding bottom is not respected 
`  <div className={cx(styles.modalContent, contentClassName)}>{children}</div>
`
**What did you expect to happen**:

  - To respect the padding when is scrolled

### 
# How do we reproduce it?

**Step 1**:

- Open a modal with scrolleable content and check it doesn't respect the padding bottom 
image
image

### 
# What Grafana version are you using?

v9.5.0

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### User's OS?

_No response_

### User's Browser?

_No response_

### Is this a Regression?

None

### Are Datasources involved?

_No response_

### Anything else to add?

_No response_

### PR DESCRIPTION
**Pull request by ivanortegaalba**
|Before|After|
|-|-|
|

**Changes**
* The list is combined into one
* The footer is fixed
* The drop area is taken into account

Fixes #67948

### CODE REVIEW

**Comment by ivanortegaalba** [comment]
Great idea! Doing some tests 😀

File: public/app/features/datasources/components/picker/DataSourceModal.tsx [hunk 1]
```diff
@@ -194,11 +196,11 @@ function getDataSourceModalStyles(theme: GrafanaTheme2) {
...
       ${theme.breakpoints.down('md')} {
         width: 100%;
-        height: 47%;
         border-right: 0;
         padding-right: 0;
-        border-bottom: 1px solid ${theme.colors.border.weak};
-        padding-bottom: ${theme.spacing(4)};
+        flex: 1;
+        // This is a hack to make the height of the list work on mobile
+        height: calc(100% - 104px);
```
Comment (juanicabanas) [inline public/app/features/datasources/components/picker/DataSourceModal.tsx:203]: is the problem that it takes too much height? 
what about doing this? 
```suggestion
        flex: 2;
        overflow-y: auto
```

File: public/app/features/datasources/components/picker/DataSourceModal.tsx [hunk 2]
```diff
@@ -212,21 +214,31 @@ function getDataSourceModalStyles(theme: GrafanaTheme2) {
 
       ${theme.breakpoints.down('md')} {
         width: 100%;
-        height: 53%;
         padding-left: 0;
-        padding-top: ${theme.spacing(4)};
+        flex: 0;
```
Comment (juanicabanas) [inline public/app/features/datasources/components/picker/DataSourceModal.tsx:218]: and we would need this to make it possible
```suggestion
        flex: 1;
```

**Comment by ivanortegaalba** [comment]
@juanicabanas In the end, I didn't change the flex, the overflow-y made it work as expected. 

| Default | With drop area |
|-|-|
Captura de pantalla 2023-05-25 a las 21 41 12|Captura de pantalla 2023-05-25 a las 21 40 31

**Review by dprokop — APPROVED** [review]

**Pull request merged**